test/com/google/enterprise/adaptor/opentext/AuthenticationHandlerTest.java

```diff
@@ -61,6 +65,63 @@ public void testGetNewToken() throws SOAPException {
     assertEquals("incoming token", handler.getAuthenticationToken());
   }
 
+  @Test
+  public void testEmptyMessage() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(false);
+    contextMock.message = null;
+    SOAPMessageContext context =
+        Proxies.newProxyInstance(SOAPMessageContext.class, contextMock);
+    AuthenticationHandler handler = new AuthenticationHandler("authtoken");
+    List<String> messages = new ArrayList<>();
+    captureLogMessages(AuthenticationHandler.class,
+        "SOAPMessageContext.message==null", messages);
+    assertTrue(handler.handleMessage(context));
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testEmptyHeaderInbound() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(false);
+    contextMock.message.getSOAPPart().getEnvelope().getHeader().detachNode();
+    assertEquals("header should be missing",
```

Comment:
This is OK, but assertNull has a improved default message now (and includes the non-null value, which it didn't before).